lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -2333,6 +2365,21 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
                     Torch::ListType::get(rewriter.getType<Torch::IntType>()),
                     padsRearrange)
                 .getResult();
+
+        // lowering to AtenConstantPadNdOp directly allows passing any torch
+        // scalar type for the value, whereas AtenPadOp takes an optional float
+        // type.
+        if (mode == "constant" &&
```

Comment:
Could is insert a `mode == "constant"` check at the copy and store as a `bool`? Makes it easiest for checking models. We can do ones for `edge` and `wrap` as well.